### PR TITLE
Making SSE event storage optional. From now the [sse_server] config s…

### PR DESCRIPTION
**Pull request by zajko**
…ection has a `disable_event_persistence` property. If set to true, Sidecar will not store events to database. [storage.sqlite_config] and [storage.postgresql_config] are optional. Also [storage.sqlite_config] and [storage.postgresql_config] have `enabled` property. If set to false, Sidecar will treat them as if they are not defined at all. Changed the name of [storage.storage_path] to [storage.storage_folder]

### CODE REVIEW

File: event_sidecar/src/event_handling_service/db_saving_event_handling_service.rs [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+        }
+        let id = sse_event.id;
+        let source = sse_event.source.to_string();
+        let api_version = sse_event.api_version;
+        let network_name = sse_event.network_name;
+        let filter = sse_event.inbound_filter;
+        let res = self
+            .database
+            .save_block_added(
+                BlockAdded::new(block_hash, block), //TODO maybe we could avoid these clones
```
Comment (rafal-ch) [inline event_sidecar/src/event_handling_service/db_saving_event_handling_service.rs:69]: Stray comment?

File: event_sidecar/src/lib.rs [hunk 2]
```diff
@@ -284,143 +304,43 @@ async fn flatten_handle<T>(handle: JoinHandle<Result<T, Error>>) -> Result<T, Er
...
-        }
-    }
-}
-
 /// Function to handle single event in the sse_processor.
 /// Returns false if the handling indicated that no other messages should be processed.
 /// Returns true otherwise.
 #[allow(clippy::too_many_lines)]
-async fn handle_single_event<Db: DatabaseReader + DatabaseWriter + Clone + Send + Sync>(
+async fn handle_single_event<EHS: EventHandlingService + Clone + Send + Sync>(
```
Comment (rafal-ch) [inline event_sidecar/src/lib.rs]: A nit, but it seems like the `Clone` bound is not needed.

**Review by rafal-ch — APPROVED** [review]
LGTM :+1:

File: event_sidecar/src/event_handling_service/db_saving_event_handling_service.rs (hunk 1 above)
Comment (zajko) [reply event_sidecar/src/event_handling_service/db_saving_event_handling_service.rs:69]: thanks :)

**Pull request merged**